### PR TITLE
Asteroids/WorldGenerator

### PR DESCRIPTION
**Pull request by relinu**
Adds a WorldGenerator, cann add asteroids or a starting area.

If this gets to many thumbs down I'll remove it ^^'

### CODE REVIEW

**Comment by relinu** [comment]
screenshot04

**Comment by vogonistic** [comment]
Nice! Maybe add a higher threshold where, if it's reached you can put a scaled stack of steel plates on there, as a placeholder for better astroid resources later. That way they aren't just pretty, they are also useful.

**Comment by ComputerOverlord** [comment]
It's great! You should add some sort of asteroid furniture (that can't be built)

**Comment by emilkris33** [comment]
@relinu Maybe you should start people out with some steelplates so they can get building

**Comment by ComputerOverlord** [comment]
@emilkris33 I think you should have to mine asteroids first.

**Comment by vogonistic** [comment]
@ComputerOverlord Mining hasn't been built yet. Plates exists and fit well as a placeholder.

**Comment by relinu** [comment]
screenshot05

So many steel plates *~*

**Comment by ComputerOverlord** [comment]
@relinu You should limit how many plates there are. 1; Maybe stacks could be smaller, 2: Halving or even dividing the amount by 4 _might_ be good...

**Comment by relinu** [comment]
Someone can adjust the values later, if you look at the code you can see that there is a min, max value of stacksize and a plate spawnchance.

**Comment by ComputerOverlord** [comment]
@relinu I figured that was the case.

**Comment by RodRitter** [comment]
I love it!

File: Assets/Scripts/Models/WorldGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+public class WorldGenerator {
+
+    public static int startAreaSize = 3;
+
+    public static TileType asteroid_floor_type = TileType.Floor;
+    public static float asteroid_noise_scale = 0.2f;
+    public static float asteroid_noise_threshhold = 0.75f;
+    public static float asteroid_ressource_chance = 0.85f;
+    public static int asteroid_ressource_min = 5;
+    public static int asteroid_ressource_max = 15;
```
Comment (OzieGamma) [inline Assets/Scripts/Models/WorldGenerator.cs]: If these are constants, I suggest using `public const int ...`

**Comment by ComputerOverlord** [comment]
@quill18 should get to this.

**Comment by quill18** [comment]
An excellent start to world generation. Merged!

**Pull request merged**